src/libs/Navigation/AppNavigator/MainDrawerNavigator.js

```diff
@@ -111,6 +115,19 @@ class MainDrawerNavigator extends Component {
         if (!this.initialParams.reportID) {
             const state = this.props.navigation.getState();
             const reportScreenKey = lodashGet(state, 'routes[0].state.routes[0].key', '');
+
+            // If this onyx key exists then it means we stashed the url from a deep link
+            // so we want to override the reportID that is being loaded by default to the one the user was attempting to load initially.
+            const reportScreenRoute = this.props.reportScreenRoute;
+            if (!_.isEmpty(reportScreenRoute)) {
+                const {reportID, isSubReportPageRoute} = ROUTES.parseReportRouteParams(reportScreenRoute);
+                if (!_.isEmpty(reportID)) {
+                    initialNextParams.reportID = reportID;
+                }
+                if (isSubReportPageRoute) {
+                    Navigation.navigate(reportScreenRoute);
```

Comment:
@sobitneupane Now, there is only one problem. That is the app doesn't redirect to screen same as Detail or Participant. I fixed it here.
About the deep link `r/:reportID`, the app redirects to the report screen because `openReportFromDeepLink` is called and it navigates to report screen with `Navigation.navigate(ROUTES.getReportRoute(report[ID));` 
https://github.com/Expensify/App/blob/2c1dfe94ce5d69a44af4a8d384903bc85a7c2e00/src/libs/actions/Report.js#L1484-L1498

Reply:
@dukenv0307 Let's continue the discussion on slack. Please go through [this slack thread](https://expensify.slack.com/archives/C01GTK53T8Q/p1682549395857859). Multiple internal engineers are suggesting to get rid of side effect behavior as discussed [here](https://github.com/Expensify/App/pull/17699#issuecomment-1526056827).